File: src/main/java/com/felipe/booking/app/entity/BookingEntity.java
package com.felipe.booking.app.entity;

import com.felipe.booking.domain.model.Booking;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(value = "booking")
@NoArgsConstructor
@AllArgsConstructor
public class BookingEntity {

    @Id
    private String id;
    private LocalDateTime checkInDate;
    private LocalDateTime checkOutDate;

    public static BookingEntity of(Booking booking) {
        return new BookingEntity(
                booking.getId(),
                booking.getCheckInDate(),
                booking.getCheckOutDate()
        );
    }

    public Booking toDomain() {
        return new Booking(
                id,
                this.checkInDate,
                this.checkOutDate
        );
    }
}

File: src/main/kotlin/com/felipe/booking/alten/app/entity/BookingEntity.kt
package com.felipe.booking.alten.app.entity

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime

@Document(value = "booking")
data class BookingEntity(
    @Id
    val id: String?,
    val checkInDate: LocalDateTime,
    val checkOutDate: LocalDateTime,
    val creationDate: LocalDateTime,
    val updateDate: LocalDateTime,
)